### PR TITLE
Fixed mission cars not being properly cleaned up.

### LINKED ISSUE #3427: [Bug]: Vehicles not being cleaned up during mission restarts
### Describe the bug

When a mission or heist is restarted (due to a player dying or an objective failing for example), your vehicle does not get removed by the game.

In the past it sometimes showed a notification saying "xyz tried to delete your vehicle", but nowadays it doesnt even show that.

**For example:** the driller failed during the Fleeca Bank job and we restarted from a checkpoint. The Armored Kuruma I was sitting in was not cleaned up by the game, and now its just sitting there as an extra unused vehicle.

car bug

### Steps To Reproduce

1. Start any mission/heist
2. Fail mid-way and restart from checkpoint
3. Your vehicle, regardless of whether you're sitting in or not, get carried over into the restart

### Expected Behavior

The game should delete them.

### Logs

[cout.log](https://github.com/user-attachments/files/16374439/cout.log)

### Store

Steam

### Language

English

### YimMenu Language

English (US)

### Additional context

I have observed this with and without my LUAs being active.

I have all my protection settings disabled (I do not play in public lobbies anyway). I have tried resetting and then disabling them again as well.

I have tried resetting my entire Yim settings as well.

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

### PR DESCRIPTION
**Pull request by gir489returns**
Closes #3427

### CODE REVIEW

**Review by Yimura — DISMISSED** [review]

File: src/hooks/protections/received_clone_create.cpp [hunk 1]
```diff
@@ -1,90 +1,93 @@
...
+		{
+			if (!NETWORK::NETWORK_IS_ACTIVITY_SESSION())     //If we're in Freemode.
+			{
+				if (auto plyr = g_player_service->get_by_id(src->m_player_id))
+				{
+					g.reactions.delete_vehicle.process(plyr);
+				}
+				return;
+			}
+		}
```
Comment (gir489returns) [inline src/hooks/protections/received_clone_create.cpp:38]: @maybegreat48 Why are we checking for a delete scenario in CLONE_CREATE?

Comment (gir489returns) [reply src/hooks/protections/received_clone_create.cpp]: This is a malicious scenario and can be only called by modders.

**Comment by gir489returns** [comment]
> The line endings have changed but LGTM otherwise

Yeah you can thank @Yimura for that amazing change to git. No matter what you do now, it will force every file to be in CR LF. We should just do a giant replace of all the files so we get it over with.

**Comment by xiaoxiao921** [comment]
If such commit happens, you can make it so that the commit doesn't affect the git blame.

**Review by Yimura — APPROVED** [review]

**Comment by gir489returns** [comment]
@Yimura The user who reported the original bug has confirmed this fixed it.

**Pull request merged**